tensorboard/plugins/projector/vz_projector/data.ts

```diff
@@ -528,12 +528,15 @@ namespace vz_projector {
           TSNE_SAMPLE_SIZE
         );
         let labels = new Array(sampledIndices.length);
-        sampledIndices.forEach(
-          (index, i) =>
-            (labels[i] = this.points[index].metadata[
-              superviseColumn
-            ].toString())
-        );
+        sampledIndices.forEach((index, i) => {
+          let label = this.points[index].metadata[superviseColumn];
+          if (!label || !label.toString()) {
+            // if rows don't match up, backfill unknown values.
+            label = 'unknown #' + index.toString();
+            this.points[index].metadata[superviseColumn] = label;
```

Comment:
We generally, for right reasons, avoid side effects like this. Do you need to mutate the object instance attached to this or can you make a copy then assign later?

Reply:
I re-wrote this section not to manipulate state, I am instead handling meta-data labels being null at the edges.